From 466d04a57b32417dbf2d5e818af1b7593af5f802 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sun, 3 May 2020 10:29:36 +0200
Subject: [PATCH] meson: add tests option

Add tests option to allow the user to build at-spi2-atk without tests,
it is especially useful as one of the test requires libxml-2.0

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status:
https://gitlab.gnome.org/GNOME/at-spi2-atk/-/merge_requests/22]
---
 meson.build       | 4 +++-
 meson_options.txt | 4 ++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 9fce5d6..61bc93d 100644
--- a/meson.build
+++ b/meson.build
@@ -61,4 +61,6 @@ install_data('at-spi2-atk.desktop',
 
 subdir('droute')
 subdir('atk-adaptor')
-subdir('tests')
+if get_option('tests')
+  subdir('tests')
+endif
diff --git a/meson_options.txt b/meson_options.txt
index 24c9228..a4abdc2 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -2,3 +2,7 @@ option('disable_p2p',
        description: 'Disable peer-to-peer DBus connections',
        type: 'boolean',
        value: false)
+option('tests',
+       description: 'Whether to build tests',
+       type: 'boolean',
+       value: true)
-- 
2.26.2

